max num lines exceeded

I'm using `exdown` in a unit testing framework. 

It's telling me my README is too long to test. How do I set `max_num_lines`?
```
____________________ ERROR collecting tests/test_README.py _____________________
tests/test_README.py:15: in <module>
    exdown.extract(this_dir.parent / "README.md", syntax_filter="python"),
.tox/py3/lib/python3.7/site-packages/exdown/main.py:3: in extract
    return from_buffer(handle, *args, **kwargs)
.tox/py3/lib/python3.7/site-packages/exdown/main.py:31: in from_buffer
    f"File too large (> {max_num_lines} lines). Set max_num_lines."
E   RuntimeError: File too large (> 10000 lines). Set max_num_lines.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -33,6 +33,5 @@ jobs:
       run: |
         pip install tox
         tox
-    - name: Submit to codecov
+    - uses: codecov/codecov-action@v1
       if: ${{ matrix.python-version == '3.8' }}
-      run: bash <(curl -s https://codecov.io/bash)

--- a/exdown/main.py
+++ b/exdown/main.py
@@ -15,8 +15,9 @@ def from_buffer(f, max_num_lines=10000, syntax_filter=None):
             # EOF
             break
 
-        if line.strip()[:3] == "```":
-            syntax = line.strip()[3:]
+        if line.lstrip()[:3] == "```":
+            syntax = line.lstrip()[3:]
+            num_leading_spaces = len(line) - len(line.lstrip())
             lineno = k - 1
             # read the block
             code_block = []
@@ -29,8 +30,12 @@ def from_buffer(f, max_num_lines=10000, syntax_filter=None):
                     raise RuntimeError(
                         f"File too large (> {max_num_lines} lines). Set max_num_lines."
                     )
-                if line.strip()[:3] == "```":
+                # check if end of block
+                if line.lstrip()[:3] == "```":
                     break
+                # Cut (at most) num_leading_spaces leading spaces
+                nls = min(num_leading_spaces, len(line) - len(line.lstrip()))
+                line = line[nls:]
                 code_block.append(line)
 
             if syntax_filter and syntax_filter.strip() != syntax.strip():

--- a/exdown/main.py
+++ b/exdown/main.py
@@ -38,7 +38,7 @@ def from_buffer(f, max_num_lines=10000, syntax_filter=None):
             if previous_line.strip() == "<!--exdown-skip-->":
                 continue
 
-            out.append(("".join(code_block[:-1]), lineno))
+            out.append(("".join(code_block), lineno))
 
         previous_line = line
 

--- a/exdown/main.py
+++ b/exdown/main.py
@@ -15,21 +15,23 @@ def from_buffer(f, max_num_lines=10000, syntax_filter=None):
             # EOF
             break
 
-        if line[:3] == "```":
-            syntax = line[3:]
+        if line.strip()[:3] == "```":
+            syntax = line.strip()[3:]
             lineno = k - 1
             # read the block
-            line = f.readline()
-            code_block = [line]
-            k += 1
-            while code_block[-1][:3] != "```":
+            code_block = []
+            while True:
                 line = f.readline()
-                code_block.append(line)
                 k += 1
+                if not line:
+                    raise RuntimeError("Hit end-of-file prematurely. Syntax error?")
                 if k > max_num_lines:
                     raise RuntimeError(
                         f"File too large (> {max_num_lines} lines). Set max_num_lines."
                     )
+                if line.strip()[:3] == "```":
+                    break
+                code_block.append(line)
 
             if syntax_filter and syntax_filter.strip() != syntax.strip():
                 continue

--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,6 @@
 [metadata]
 name = exdown
-version = 0.7.0
+version = 0.7.1
 author = Nico Schlömer
 author_email = nico.schloemer@gmail.com
 description = Extract code blocks from markdown

--- a/test/example.md
+++ b/test/example.md
@@ -9,11 +9,11 @@ import exdown
 
 exdown.from_buffer
 ```
-Something that intentionally does not work:
+Something that should be skipped because of the language:
 ```bash
 foobar
 ```
-Again with syntax
+Again with an explicit skip:
 <!--exdown-skip-->
 ```python
 bar
@@ -23,3 +23,8 @@ Something that contains triple fences
 ```python
 # ```import math```
 ```
+
+Indented code blocks:
+  ```python
+  1 + 1 == 2
+  ```

--- a/test/example.md
+++ b/test/example.md
@@ -28,3 +28,8 @@ Indented code blocks:
   ```python
   1 + 1 == 2
   ```
+
+"Wrong" indentation:
+```python
+1 + 1 == 2
+  ```

--- a/test/test_exdown.py
+++ b/test/test_exdown.py
@@ -22,9 +22,30 @@ def test_string(string, lineno):
     exec(string)
 
 
+# example.md against reference strings test against
+def test_reference():
+    ref = [
+        ("1 + 2 + 3\n2 + 5\n", 2),
+        ("import exdown\n\nexdown.from_buffer\n", 7),
+        ("# ```import math```\n", 23),
+        ("1 + 1 == 2\n", 28),
+        ("1 + 1 == 2\n", 33),
+    ]
+    lst = exdown.extract(this_dir / "example.md", syntax_filter="python")
+    for r, obj in zip(ref, lst):
+        assert r == obj
+
+
 @pytest.mark.parametrize(
     "string, lineno",
     exdown.extract(this_dir / "example.md", syntax_filter="python"),
 )
 def test_file(string, lineno):
     exec(string)
+
+
+if __name__ == "__main__":
+    blocks = exdown.extract(this_dir / "example.md", syntax_filter="python")
+    for block in blocks:
+        print(block)
+        print()

